src/els_elvis_diagnostics.erl

```diff
@@ -0,0 +1,83 @@
+%%==============================================================================
+%% Compiler diagnostics
+%%==============================================================================
+-module(els_elvis_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+-export([ diagnostics/1 ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+-spec diagnostics(uri()) -> [diagnostic()].
+diagnostics(Uri) ->
+  RelFile = els_utils:project_relative(Uri),
+
+  %% Note: elvis_core:rock_this requires a file path relative to the
+  %%       project root, formatted as a string.
+  WS = try elvis_core:rock_this(RelFile)
+       catch Err ->
+           lager:info("Elvis error.[Err=~p] ", [Err]),
+           ok  %% TODO: return a diagnostic saying elvis failed?
+       end,
+  case WS of
+      ok -> [] ;
+      {fail, Problems} -> lists:flatmap(fun format_diagnostics/1, Problems)
+  end.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+-spec format_diagnostics(any()) -> [map()].
+format_diagnostics(#{file := Path, rules := Rules}) ->
+  R = format_rules(Path, Rules),
+  lists:flatten(R).
+
+
+%%% This section is based directly on elvis_result:print_rules
```

Comment:
@jfacorro Most of the complexity in this PR seems to come from this function and the relative path handling. Would it make sense to add those to elvis itself? I.e. allowing absolute paths and separating formatting from outputting?

Reply:
We would have to check how that impacts elvis code, since the reason I think the paths are relative is because it simplifies reporting the path when printing.

Reply:
@jfacorro Should we then proceed with this PR as it is and look at the elvis changes separately, then?

Reply:
Yes, I don't think we should wait on changes on elvis, if the code in this PR works.